### PR TITLE
Rework file config io

### PR DESCRIPTION
**Pull request by vallsv**
This isolate io for file config into the `io` package. It remove duplicated code and provide a safer compatibility with older config version (while dict are mess).

- Isolate `ponifile`read/write (we could do more: ideally `get/set_config` from azimuthalInteg should not contain `version`, as version should only be part of the storage -> the file)
- helper for integration-config (the json file).

### CODE REVIEW

File: pyFAI/worker.py [hunk 1]
```diff
@@ -413,6 +380,8 @@ def process(self, data, variance=None, normalization_factor=1.0, writer=None, me
         if self.azimuth_range is not None:
             kwarg["azimuth_range"] = self.azimuth_range
 
+        print(kwarg)
```
Comment (kif) [inline pyFAI/worker.py]: Trailing print !

File: pyFAI/worker.py [hunk 2]
```diff
@@ -519,6 +489,7 @@ def set_config(self, config, consume_keys=False):
             filenames = _read_filenames(filename)
             method = "mean"
             data = _reduce_images(filenames, method=method)
+            print(data)
```
Comment (kif) [inline pyFAI/worker.py]: trailing print

File: pyFAI/worker.py [hunk 3]
```diff
@@ -529,6 +500,7 @@ def set_config(self, config, consume_keys=False):
...
```
Comment (kif) [inline pyFAI/worker.py]: trailing print

**Comment by vallsv** [comment]
Fixed

**Pull request merged**